tests/test_tasks/test_task.py

```diff
@@ -42,16 +43,30 @@ def test_download_task(self):
 
     def test_upload_task(self):
 
-        dataset_id = self._get_compatible_rand_dataset()
-        # TODO consider implementing on the diff task types.
-        task = create_task(
-            task_type_id=self.task_type_id,
-            dataset_id=dataset_id,
-            target_name=self._get_random_feature(dataset_id),
-            estimation_procedure_id=self.estimation_procedure
-        )
+        # We don't know if the task in question already exists, so we try a few times. Checking
+        # beforehand would not be an option because a concurrent unit test could potentially
+        # create the same task and make this unit test fail (i.e. getting a dataset and creating
+        # a task for it is not atomic).
+        for i in range(100):
+            try:
+                dataset_id = self._get_compatible_rand_dataset()
+                # TODO consider implementing on the diff task types.
+                task = create_task(
+                    task_type_id=self.task_type_id,
+                    dataset_id=dataset_id,
+                    target_name=self._get_random_feature(dataset_id),
+                    estimation_procedure_id=self.estimation_procedure
+                )
+
+                task_id = task.publish()
+                # success
+                break
+            except OpenMLServerException as e:
+                if e.code == 614:
+                    continue
+                else:
+                    raise e
```

Comment:
Maybe use an `else` statement (as part of [`for` `else`](http://book.pythontips.com/en/latest/for_-_else.html#else-clause)) to fail the test explicitly with according descriptive message (rather than having to go from `task_id not defined`).

Reply:
Great idea! I just added this.